src/terrain/services/filesystem/directory.clj

```diff
@@ -146,90 +148,13 @@
-                          :offset      offset
-                          :bad-chars   (cfg/fs-bad-chars)
-                          :bad-name    (cfg/fs-bad-names)
-                          :bad-path    (bad-paths user)
-                          :sort-field  sort-field
-                          :sort-dir    sort-dir}
-        params           (if info-type
-                           (assoc params :info-type info-type)
-                           params)
-        handle-not-found (fn [_ _ _] (throw+ {:error_code error/ERR_NOT_FOUND :path path}))]
-    (data/trapped-request :get url-path {:query-params params}
-      :403 handle-not-found
-      :404 handle-not-found
-      :410 handle-not-found
-      :414 handle-not-found
-      :422 handle-not-processable
-      :503 handle-error-with-rethrow)))
-
-
-(defn- resolve-sort-field
-  [sort-col]
-  (if-not sort-col
-    "name"
-    (case (string/lower-case sort-col)
-      "datecreated"  "datecreated"
-      "id"           "path"
-      "lastmodified" "datemodified"
-      "name"         "name"
-      "path"         "path"
-      "size"         "size"
-                     (do
-                       (log/warn "invalid sort column" sort-col)
-                       (throw+ {:error_code "ERR_INVALID_SORT_COLUMN" :column sort-col})))))
-
-
-(defn- resolve-sort-dir
-  [sort-dir]
-  (if-not sort-dir
-    "ASC"
-    (let [sort-dir (string/upper-case sort-dir)]
-      (when-not (contains? #{"ASC" "DESC"} (string/upper-case sort-dir))
-        (log/warn "invalid sort order" sort-dir)
-        (throw+ {:error_code "ERR_INVALID_SORT_DIR" :sort-dir sort-dir}))
-      sort-dir)))
-
-
-; TODO validate limit >= 0, offset >= 0
 (defn do-paged-listing
   "Entrypoint for the API that calls (paged-dir-listing)."
-  [{:keys [user path entity-type info-type limit offset sort-col sort-dir]}]
-  (Integer/parseInt limit)
-  (Integer/parseInt offset)
-  (let [path        (ft/rm-last-slash path)
-        entity-type (duv/resolve-entity-type entity-type)
-        sort-field  (resolve-sort-field sort-col)
-        sort-dir    (resolve-sort-dir sort-dir)
-        resp        (paged-dir-listing
-                      user path entity-type limit offset sort-field sort-dir info-type)]
-    (format-page user (json/decode (:body resp) true))))
-
-(with-pre-hook! #'do-paged-listing
-  (fn [params]
-    (paths/log-call "do-paged-listing" params)
-    (validate-map params {:user string? :path string? :limit string? :offset string?})))
-
-(with-post-hook! #'do-paged-listing (paths/log-func "do-paged-listing"))
+  [{user :shortUsername} {:keys [path] :as params}]
+  (let [params (dissoc params :path)]
+    (->> (merge params {:user      (or user "anonymous")
+                        :bad-chars (cfg/fs-bad-chars)
+                        :bad-name  (cfg/fs-bad-names)
+                        :bad-path  (bad-paths user)})
+         (data/list-folder-contents path)
+         (format-page user))))
```

Comment:
Getting rid of the error handlers has the potential to break a few QA tests, but I think the results will be worth the test breakages. The primary result is that errors returned by `data-info` will be returned to the caller as-is without any alteration. For example, doing a listing for a directory that doesn't exist used to return a response code of `500` along with a message saying that the service call to `data-info` failed:

```
$ curl -sH "$AUTH_HEADER" "https://qa.cyverse.org/terrain/secured/filesystem/paged-directory?path=/iplant/home/sarahrrrr&limit=1000&offset=0" | jq .
{
  "error_code": "ERR_REQUEST_FAILED",
  "message": "GET http://data-info/data/path/iplant/home/sarahrrrr had a service error: internal error"
}
```

Now it will continue to return a status code of 500, but the error will indicate that the path doesn't exist:

```
$ curl -sH "$AUTH_HEADER" "http://localhost:31325/filesystem/paged-directory?path=/iplant/home/sarahrrrr&limit=1000" | jq .
{
  "error_code": "ERR_DOES_NOT_EXIST",
  "path": "/iplant/home/sarahrrrr"
}
```

There were some error handlers designed to catch errors for missing file paths in `terrain`, but they were never being called because `data-info` returns a status code of `500`. It should probably be updated to return a `404` at some point, but this error is better than what we were getting before. Also, modifying `data-info` to return a `404` in that case won't require any changes to terrain.

I also removed a lot of the query parameter validation, primarily because `compojure-api` does a lot of it for us. Any remaining query parameter validation can be handled by `data-info` when it gets the request.